pkg/networkservice/common/authorize/server.go

```diff
@@ -33,63 +30,35 @@ import (
 )
 
 type authorizeServer struct {
-	p *rego.PreparedEvalQuery
+	policies []opa.AuthorizationPolicy
 }
 
 // NewServer - returns a new authorization networkservicemesh.NetworkServiceServers
-func NewServer(p *rego.PreparedEvalQuery) networkservice.NetworkServiceServer {
+func NewServer(policies ...opa.AuthorizationPolicy) networkservice.NetworkServiceServer {
```

Comment:
Here I suggest using a set of policies when can be used default policies such as `tokens expired policy`, `tokens valid`. 
Example: 
```go
s := next.NewServer(authorize.NewServer(opa.WithTokensExpiredPolicy(), opa.WithSomePolicy()...))
```
@haiodo , @edwarnicke , @fkautz  What do you think? Please share your thoughts.

Reply:
I like where this is going *a lot*.

When you bubble this up to the endpoint and client chains.. you could basically then let folks pass in an 'authzServer' or 'authzClient' into the constructors for those chains :)